src/Language/Visitor.php

```diff
@@ -217,12 +217,23 @@ public static function visit(object $root, array $visitor, ?array $keyMap = null
                             $node->splice($editKey, 1);
                             ++$editOffset;
                         } elseif ($node instanceof NodeList) {
-                            if (! $editValue instanceof Node) {
+                            if ($editValue instanceof NodeList) {
+                                // Replace a single node with multiple nodes from a NodeList
+                                // We need to splice the NodeList's contents into the parent
+                                $replacementNodes = [];
+                                foreach ($editValue as $replacementNode) {
+                                    $replacementNodes[] = $replacementNode;
+                                }
+                                // Remove the original node and insert all replacement nodes
+                                $node->splice($editKey, 1, $replacementNodes);
```

Comment:
Let's make `NodeList::splice` more flexible and have it accept any `iterable` instead of `array`.